tests/integration/ha_tests/test_self_healing.py

```diff
@@ -1,61 +1,107 @@
 #!/usr/bin/env python3
 # Copyright 2022 Canonical Ltd.
 # See LICENSE file for licensing details.
+import logging
 
 import pytest
 from pytest_operator.plugin import OpsTest
 from tenacity import Retrying, stop_after_delay, wait_fixed
 
+from tests.integration.ha_tests.conftest import APPLICATION_NAME
 from tests.integration.ha_tests.helpers import (
+    METADATA,
+    check_writes,
     count_writes,
     fetch_cluster_members,
     get_primary,
     is_replica,
     postgresql_ready,
     secondary_up_to_date,
     send_signal_to_process,
-    stop_continuous_writes,
+    start_continuous_writes,
 )
-from tests.integration.helpers import app_name, build_and_deploy, get_unit_address
+from tests.integration.helpers import (
+    CHARM_SERIES,
+    app_name,
+    build_and_deploy,
+    get_unit_address,
+)
+
+logger = logging.getLogger(__name__)
 
-PATRONI_PROCESS = "/usr/local/bin/patroni"
+APP_NAME = METADATA["name"]
+PATRONI_PROCESS = "/usr/bin/patroni"
 POSTGRESQL_PROCESS = "postgres"
```

Comment:
We are likely hitting the same pkill issues as the VM charm here, though we may want to deal with that in a separate PR

Reply:
Good catch! I'll check it and update the code.

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/134/files/b79aa19b4a9f197c575d25345c780b47bb9f2d0d..c29e5cd37540872c7e19973598d8c77576d5388a. I used the same logic that we have in the VM charm (SIGINT). I'm going to open another PR to reenable the PG process freeze db test.